### ISSUE TITLE
Improvements to built-in TypeScript support 

### ISSUE THREAD

**Issue by jason0x43** [theintern/intern#1000]
The built in TS support makes working with TS much simpler, but it has problems when a test process is started with ts-node. This might be done if a user has written a custom testing script in TS, or during the development of Intern itself when Intern is run to test itself.

When running a script with ts-node that imports and executes intern programmatically, Intern tries to use TS node to process modules that the external ts-node has already processed. Since the external ts-node has already transpiled the code and stripped out the type annotations, Intern's ts-node handler breaks.

The current workaround is to manually disable Intern's built-in TS support by setting the `node.tsconfig ` property to `false`.

## Expected behavior

Ideally, Intern would detect that a ts-node loader was already active and wouldn't try to double-transpile files.

## Environment

Intern version: 4.7.0
Node version: 13.3
NPM version: 6.13.2

**Comment by maier49** [comment]
I think the fundamental issue here is not that the modules are transpiled twice, but that ts-node is being registered twice and the second time (by Intern) it's being registered with inadequate configuration due it lacking `tsconfig-paths/register`. I've verified this by testing that intern still works with a dojo app even if it also uses the existing tsnode plugin to register ts-node twice (but with the same config as intern), and by testing that the repo-reunion branch works after having removed the `ts-config: false` flag as long as it doesn't register ts-node. 

I've opened up a PR that attempts to check for a registered ts-node, but unfortunately unless TypeStrong/ts-node#409 gets a solution this check isn't going to be completely reliable.
